File: backend/app/schemas/advisor.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


MessageRole = Literal["user", "assistant"]


class AdvisorMessage(BaseModel):
    role: MessageRole
    content: str
    createdAt: datetime

    model_config = ConfigDict(from_attributes=True)


class AdvisorSessionResponse(BaseModel):
    sessionId: str
    createdAt: datetime
    messages: list[AdvisorMessage]

    model_config = ConfigDict(from_attributes=True)


class AdvisorMessageRequest(BaseModel):
    message: str = Field(min_length=1)


class AdvisorMessageResponse(BaseModel):
    sessionId: str
    reply: AdvisorMessage

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/advisor.ts
import { apiRequest } from "./client";

export interface AdvisorMessage {
  role: "user" | "assistant";
  content: string;
  createdAt: string;
}

export interface AdvisorSession {
  sessionId: string;
  createdAt: string;
  messages: AdvisorMessage[];
}

export interface AdvisorMessageResponse {
  sessionId: string;
  reply: AdvisorMessage;
}

export async function createAdvisorSession(): Promise<AdvisorSession> {
  return apiRequest<AdvisorSession>("/advisor/sessions", {
    method: "POST",
  });
}

export async function getAdvisorSession(sessionId: string): Promise<AdvisorSession> {
  return apiRequest<AdvisorSession>(`/advisor/sessions/${sessionId}`);
}

export async function sendAdvisorMessage(
  sessionId: string,
  message: string,
): Promise<AdvisorMessageResponse> {
  return apiRequest<AdvisorMessageResponse>(`/advisor/sessions/${sessionId}/messages`, {
    method: "POST",
    json: { message },
  });
}
